core/src/main/java/org/apache/calcite/sql/SqlHexCharStringLiteral.java

```diff
@@ -33,16 +34,30 @@
   final HexCharLiteralFormat format;
   final CharacterSet charSet;
 
+  public SqlHexCharStringLiteral(final String hex, final SqlParserPos pos,
```

Comment:
Now that I look at this more I think it makes more sense to create a new class to deal with bytes, thoughts?

Reply:
Especially with the differences between unparsing a byte literal and a regular hex string.

Reply:
Yes, I think a separate byte literal class sounds like a good idea.

Reply:
I've made the changes.